staging/src/k8s.io/kubectl/pkg/cmd/diff/prune.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diff
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/cli-runtime/pkg/resource"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"k8s.io/kubectl/pkg/util/prune"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/dynamic"
+)
+
+type pruner struct {
+	mapper        meta.RESTMapper
+	dynamicClient dynamic.Interface
+
+	visitedUids       sets.String
+	visitedNamespaces sets.String
+	labelSelector     string
+	resources         []prune.Resource
+}
+
+func newPruner(dc dynamic.Interface, m meta.RESTMapper, r []prune.Resource) *pruner {
+	return &pruner{
+		visitedUids:       sets.NewString(),
+		visitedNamespaces: sets.NewString(),
+		dynamicClient:     dc,
+		mapper:            m,
+		resources:         r,
+	}
+}
+
+func (p *pruner) pruneAll() ([]runtime.Object, error) {
```

Comment:
That is a good idea. However, apply pruning works the way whenever it gets error, it returns. On the other hand, diff prune works different, it ignores the errors and walks through all objects to show all possible prunes. 

Although, they are doing the same thing, functionality slightly differs, since apply/prune is transactional and diff/prune is just guidance.